transformer/transform.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/local/bin/python3.6
+import json
+import logging
+import sys
+import typing
+import uuid
+from collections import namedtuple
+
+import argparse
+
+
+"""A metadata link is all the information necessary to get metadata
+for a particular field from the field that links to it."""
+MetadataLink = namedtuple('MetadataLink', ['source_field_name', 'linked_field_name', 'link_name'])
+
+
+class Transformer:
+
+    @staticmethod
+    def _build_data_objects_dict(data_objects: list):
+        return {entry['id']: entry for entry in data_objects}
```

Comment:
I'm considering switching the format (back) to this. Good to see it will be a painless change for you here.

It has the benefit of being able to address items by ID when traversing the metadata.